### ISSUE TITLE
[KIC 2.0] compose KIC deployment yaml

### ISSUE THREAD

**Issue by ccfishk** [Kong/kubernetes-ingress-controller#1480]
What this PR does / why we need it:
Need compose KIC deployment yaml which specify namespace, configmap etc.

Which issue this PR fixes **: fixes #

This PR is to compose a controller yaml file for KIC2.0 release

Special notes for your reviewer:

PR Readiness Checklist:

Complete these before marking the PR as ready to review:

**Comment by shaneutt** [comment]
@ccfishk we currently support a `helm` based deployment of the KIC and kustomize, which are both accounted for in the KIC 2.0 scope. Is the purpose of this to add a single manifest deployment option? There's plenty of other tools that do this and I'm not opposed to this, but if this is the case then I believe we should tag this as an `enhancement/feature` and _not add it as a blocker for KIC 2.0 beta_ as this will increase the scope and I haven't (personally) seen any strong end-user demand recently for this additional deployment mechanism.

**Comment by ccfishk** [comment]
@shaneutt  helm deployment is good. 

Meanwhile, another k8s generic deployment way is using yaml file as. 
i.e
k apply -f resource-kic.yaml (including CRD, role/rolebinding, rbac etc)

**Comment by shaneutt** [comment]
> @shaneutt helm deployment is good.
> 
> Meanwhile, another k8s generic deployment way is using yaml file as.
> i.e
> k apply -f resource-kic.yaml (including CRD, role/rolebinding, rbac etc)

Understood: so this is a _new functionality request_ yes? So should we put this in the `POST KIC 2.0` milestone?

**Comment by ccfishk** [comment]
> > @shaneutt helm deployment is good.
> > Meanwhile, another k8s generic deployment way is using yaml file as.
> > i.e
> > k apply -f resource-kic.yaml (including CRD, role/rolebinding, rbac etc)
> 
> Understood: so this is a _new functionality request_ yes? So should we put this in the `POST KIC 2.0` milestone?

Yes, the yaml as well test against yaml and image should be added as most users would to use as how they use 1.3.x

**Comment by ccfishk** [comment]
we will continue using helm charts deployment in 2.0
